rustls/src/common_state.rs

```diff
@@ -25,62 +27,35 @@ use pki_types::CertificateDer;
 
 /// Connection state common to both client and server connections.
 pub struct CommonState {
-    pub(crate) negotiated_version: Option<ProtocolVersion>,
-    pub(crate) side: Side,
-    pub(crate) record_layer: record_layer::RecordLayer,
-    pub(crate) suite: Option<SupportedCipherSuite>,
-    pub(crate) alpn_protocol: Option<Vec<u8>>,
-    pub(crate) aligned_handshake: bool,
-    pub(crate) may_send_application_data: bool,
-    pub(crate) may_receive_application_data: bool,
-    pub(crate) early_traffic: bool,
-    sent_fatal_alert: bool,
-    /// If the peer has signaled end of stream.
-    pub(crate) has_received_close_notify: bool,
-    pub(crate) has_seen_eof: bool,
-    pub(crate) received_middlebox_ccs: u8,
-    pub(crate) peer_certificates: Option<Vec<CertificateDer<'static>>>,
-    message_fragmenter: MessageFragmenter,
     pub(crate) received_plaintext: ChunkVecBuffer,
     sendable_plaintext: ChunkVecBuffer,
     pub(crate) sendable_tls: ChunkVecBuffer,
-    queued_key_update_message: Option<Vec<u8>>,
 
-    #[allow(dead_code)] // only read for QUIC
-    /// Protocol whose key schedule should be used. Unused for TLS < 1.3.
-    pub(crate) protocol: Protocol,
-    #[cfg(feature = "quic")]
-    pub(crate) quic: quic::Quic,
-    pub(crate) enable_secret_extraction: bool,
+    inner: CommonStateInner,
+}
+
+impl ops::DerefMut for CommonState {
```

Comment:
Hmm, I don't love extending the deref tower with another layer. What do you think about moving in the other direction, wrapping up `received_plaintext`, `sendable_plaintext` and `sendable_tls` into a `CommonBuffers` type and sticking that in the current `Connection` types as a sibling of `common`?